ocaml/idl/datamodel.ml

```diff
@@ -6663,17 +6663,44 @@ let vdi_list_changed_blocks = call
+      ~name:_vdi_nbd_server_info
+      ~descr:"Details for connecting to a VDI using the Network Block Device protocol"
+      ~gen_events:false
+      ~messages:[]
+      ~doccomments:[]
+      ~messages_default_allowed_roles:(Some []) (* No messages, so no roles allowed to use them *)
+      ~contents:
+      [ (* uid _vdi_nbd_server_info; The uuid is not needed here and only adds inconvenience. *)
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "exportname" "The exportname to request over NBD. This holds details including an authentication token, so it must be protected appropriately. Clients should regard the exportname as an opaque string or token.";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "address" "An address on which the server can be reached; this can be IPv4, IPv6, or a DNS name.";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:Int "port" "The TCP port";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "cert" "The TLS certificate of the server";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "subject" "For convenience, this redundant field holds a subject of the certificate.";
+      ] ()
+end
+let vdi_nbd_server_info = Vdi_nbd_server_info.vdi_nbd_server_info
+
 let vdi_get_nbd_info = call
     ~name:"get_nbd_info"
     ~in_oss_since:None
     ~in_product_since:rel_inverness
-    ~params:[Ref _vdi, "self", "The VDI to access via NBD."]
+    ~params:[Ref _vdi, "self", "The VDI to access via Network Block Device protocol"]
     ~errs: [Api_errors.vdi_incompatible_type]
-    ~result:(Set String, "The list of URIs.")
-    ~doc:"Get a list of URIs specifying how to access this VDI via the NBD server of XenServer. A URI will be returned for each PIF of each host that is connected to the VDI's SR. An empty list is returned in case no network has a PIF on a host with access to the relevant SR. To access the given VDI, any of the returned URIs can be passed as the export name to the nbd-server running at the IP address and port specified by that URI."
+    ~result:(Set (Record _vdi_nbd_server_info), "The details necessary for connecting to the VDI over NBD. This includes an authentication token, so must be treated as sensitive material and must not be sent over insecure networks.")
+    ~doc:"Get details specifying how to access this VDI via a Network Block Device server. For each of a set of NBD server addresses on which the VDI is available, the return value set contains a vdi_nbd_server_info object that contains an exportname to request once the NBD connection is established, and connection details for the address. An empty list is returned if there is no network that has a PIF on a host with access to the relevant SR, or if no such network has been assigned an NBD-related purpose in its purpose field. To access the given VDI, any of the vdi_nbd_server_info objects can be used to make a connection to a server, and then the VDI will be available by requesting the exportname."
```

Comment:
Maybe it would make the doc even clearer if we said somehow that the export name is from the vdi_nbd_info object they picked:
"and then the VDI will be available by requesting the exportname." -> "and then the VDI will be available by requesting the exportname specified by that object vdi_nbd_server_info."
But what is and isn't clear is quite subjective though.